packages/wrangler/src/pages/functions.tsx

```diff
@@ -0,0 +1,98 @@
+import { existsSync, lstatSync, readFileSync, writeFileSync } from "node:fs";
+import path from "node:path";
+import { FatalError } from "../errors";
+import { logger } from "../logger";
+import { isInPagesCI, ROUTES_SPEC_VERSION } from "./constants";
+import {
+	isRoutesJSONSpec,
+	optimizeRoutesJSONSpec,
+} from "./functions/routes-transformation";
+import { pagesBetaWarning } from "./utils";
+import type { ArgumentsCamelCase, Argv } from "yargs";
+
+type OptimizeRoutesArgs = {
+	"routes-path": string;
+	"output-routes-path": string;
+};
+
+export function OptimizeRoutesOptions(yargs: Argv): Argv<OptimizeRoutesArgs> {
```

Comment:
In my PR, I updated our commands' types to be inferred from the options factory return type. See https://github.com/cloudflare/wrangler2/pull/1642/files#diff-6cf5a21278239fb871170f60ab7aacc6d2ba39f0aa54c3e22db651094f4b1fb9L32

So you could do something like this:

```suggestion
import type { YargsOptionsToInterface } from "./types";
import type { Argv } from "yargs";

type OptimizeRoutesArgs = YargsOptionsToInterface<typeof OptimizeRoutesOptions>;

export function OptimizeRoutesOptions(yargs: Argv) {
```